array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,13 @@
+/* exported compact */
+
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] !== false && array[i] !== null && !Number.isNaN(array[i]) &&
```

Comment:
This does work! But we can simplify it by only checking if the value of the current index item is truthy since we just want to eliminate falsy values:
```javascript
if(array[i]) {
newArray.push(array[i]
}
```

Reply:
ah so just chuck the array[i] in the condition and call it a day then?